opensearch_py_ml/ml_models/sentencetransformermodel.py

```diff
@@ -1028,8 +1022,43 @@ def make_model_config_json(
         if model_name is None:
             model_name = self.model_id
 
-        # if user input model_type and embedding_dimension, it will skip reading the config.json file
-        if model_type is None or embedding_dimension is None:
+        # if user input model_type/embedding_dimension/pooling_mode, it will skip this step.
+        model = SentenceTransformer(self.model_id)
```

Comment:
can we use `model_name` here? Customer might provide another model name here

Reply:
I can change it to be `model_name`, but `model_name` still needs to be hugging face `model_id` for it to work though (i.e., 'sentence-transformers/msmarco-distilbert-base-tas-b'). 

I was also wondering why we need to have model_name. I thought that we just want to allow user to set the arbitrary name for the model. But in that case line 1026 should still be `self.model_id` because `SentenceTransformer("Arbitrary Name")` would not work [Note that it is hugging face class `SentenceTransformer` not our class ` SentenceTransformerModel` here.] I'm not sure if I misunderstand anything. Let me know if you still want me to change it to `model_name`.

Reply:
You were right. We can't use `model_name`. For some reason I mixed up with `model_name` with `model_id`. Ignore my comment.